### ISSUE TITLE
Problem with global sequences

### ISSUE THREAD

**Issue by altipeak** [2ndQuadrant/bdr#176]
Hi, I've discovered that if you convert sequences to be golbal on an existing database, if the current value of the sequence on a table is greater than 1000 and one node is assigned to use the range of (0,1000], then your sequence value will be updated to 1 resulting in a duplicated id error.

How to test it.
1- Create a database with a table with incrementar index id.
2- Insert more than 1000 rows.
3 - Configure a multimaster cluster of 2 nodes.
4 - alter sequence of the table id to use bdr (global sequence)

Check if the selected range if values to be used is 1-1000

5 - Insert a row on the node who has the 1-1000 range assigned to the value.
The result will be a duplicated id.

Note: If the database is a new one and you insert the rows after configure it, will work fine.

**Comment by ringerc** [comment]
Yeah, support for converting normal sequences to global sequences and retaining values is somewhere between rudimentary and nonexistent. It's only allowed at all as a workaround for issues with dump and restore.

We should probably insert fake chunks into the sequence elections and voting tables to reserve the normal sequence's original value. But that's still going to be imperfect. If you change it back to a normal sequence, reset it to 1, then turn it back into a global sequence BDR will ignore the change and carry on where it left off. Well, mostly. It won't restart at the beginning like if you drop and recreate a normal sequence, it'll acquire the next unclaimed chunks as if it was a new node.

While this is a bug, I don't expect to be able to work on it in the near future. The architecture of global sequences has a number of issues, given that it `ERROR`s when sequences are exhausted rather than blocking, it doesn't work when running on a single standalone node, etc. For 9.6bdr we'll be approaching the problem fairly differently, and the underlying sequence support in 9.4bdr doesn't allow us to fix some of the issues without major rework.

Right now I think the main thing to do is document this as a caveat.

**Comment by ringerc** [comment]
@PJMODOS  noted in chat that we could probably just use the `last_value` of an existing sequence as the startval if none is specified.

**Comment by changsijay** [comment]
I have a table which record login events named `audit_events` 
and there is a sequence named `audit_events_id_seq`

I configured 2 BDR node and used dumped sql file to restore into 1 node.
2 node data are matched now.

But after I login the web app(gitlab), it added 1 row into `audit_events` 
but the `audit_events_id_seq` last_value only increased in the node which I login in.
another node `audit_events_id_seq` didn't change.

Is it the same issue mentioned here?
However, the table is not that important, I tried:

```
delete from audit_events;
select setval(''audit_events_id_seq", 1);
```
still have the issue, 

Any way I can make `audit_events_id_seq` synced automatically?
thanks :)

**Comment by ringerc** [comment]
CSJ, if you can setval(...) then it's not a BDR global sequence.​ Since you
restored from a dump, did you set the default sequence type to 'bdr' or
ALTER SEQUENCE ... USING 'bdr' on the sequence?

**Comment by ringerc** [comment]
altipeak, currently conversion to global sequences isn't very bright. It
doesn't really take account of existing values very well, and is certainly
an area that could use improvement.

**Comment by changsijay** [comment]
Hi @ringerc , below are my steps

- dump the webapp whole database `gitlabhq_production` from original postgresql(no BDR).
- prepared 2 machines with postgresql-bdr-9.4 installed.
- create an empty database `gitlabhq_production` and some users account on 2 machines
- follow the [the quick guide](https://github.com/agios/docker-postgres-bdr) set up BDR nodes
- restore the dump into 1 node and found the tables are synced to another node.

I have no idea about how to modify a sequence to use bdr.
wondering if all tables/sequences in the `gitlabhq_production` database using bdr by above steps?

**Comment by changsijay** [comment]
below is the commands related to `audit_events_id_seq` in the dump file.

```
CREATE SEQUENCE audit_events_id_seq
    START WITH 1
    INCREMENT BY 1
    NO MINVALUE
    NO MAXVALUE
    CACHE 1;

ALTER TABLE public.audit_events_id_seq OWNER TO gitlab;
ALTER SEQUENCE audit_events_id_seq OWNED BY audit_events.id;
ALTER TABLE ONLY audit_events ALTER COLUMN id SET DEFAULT nextval('audit_events_id_seq'::regclass);
SELECT pg_catalog.setval('audit_events_id_seq', 277, true);
```

**Comment by aletabu** [comment]
Hi changsijay, it seems you need to use global sequences instead of the plain ones.
try "ALTER SEQUENCE audit_events_id_seq USING  bdr" in one of your nodes"

**Comment by changsijay** [comment]
It works find now, thanks a lot, 
sorry to annoying different issue inside this issue..
